crates/ursa-index-provider/src/engine.rs

```diff
@@ -208,7 +212,7 @@ where
         let peer_id = PeerId::from(self.provider.keypair().public());
 
         let listener_addresses = listener_addresses_receiver.await?;
```

Comment:
Do we need to advertised these other addresses that we're getting for the swarm? @theBeardA

Reply:
I don't think so; we only need to advertise where the content is retrievable, which is the rpc server.

Reply:
The indexers filters out all the local addresses. So there is no problem adding them to the list. When ran locally, most of the time public address is not available, so this was added for that. The local instance of store the index does not filter local addresses

Reply:
> The local instance of store the index does not filter local addresses
But it can be a problem for the local gateway (prod is fine), as they got swarm addresses in the provider result list.

Reply:
#158 we try all provider addresses until we run out of addresses, get an error, get a 404 or a valid response

Reply:
Sound good. 👍

Reply:
this depends on the design. the idea of the storetheindex was to advertise p2p address along with transport protocol(bitswap or graphsync) where peers can get data from. like it was one piece of content routing on the p2p(cache) node. But querying from the gateway, now we need server address. @b0xtch

Reply:
Let me clarify. We pick one provider and try all of its advertised addresses (until we run out of addresses, get a 404 or any other response). 

Updated tests.

Reply:
https://github.com/fleek-network/ursa/issues/187